### PR TITLE
Clicking a DOI link in the preview pane no longer crashes

### LINKED ISSUE #7706: DOI link: opens in preview and causes an exception
JabRef 5.3--2021-05-05--a3f093d
Linux 4.19.0-16-amd64 amd64 
Java 16.0.1 
JavaFX 16+8

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/main and the problem persists

When clicking on the DOI link displayed in the preview panel, the webpage is displayed in the preview (which is unexpected), and barely unusable (because it is very small), and then an exception is raised.

1. Open the file jabref-authors.bib (https://github.com/JabRef/jabref/blob/main/src/test/resources/testbib/jabref-authors.bib)
2. Double-click on the entry Afzal et al. (for example). The entry editor opens.
3. In the preview panel, click on the DOI

Note: the exception is surely a bug. But is the display of the website in the preview panel a new feature? If so, how to get the previous behaviour back? (i.e. opening links in the regular web browser)

<details>
  <summary>Uncaught exception details</summary>
  
  ```
netscape.javascript.JSException: ReferenceError: Can't find variable: Mark
	at org.jabref.merged.module@5.3.307/com.sun.webkit.dom.JSObject.fwkMakeException(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.WebPage.twkExecuteScript(Native Method)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.WebPage.executeScript(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.scene.web.WebEngine.executeScript(Unknown Source)
	at org.jabref@5.3.307/org.jabref.gui.preview.PreviewViewer.highlightSearchPattern(Unknown Source)
	at org.jabref@5.3.307/org.jabref.gui.preview.PreviewViewer.lambda$new$1(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.beans.property.ReadOnlyObjectPropertyBase.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.beans.property.ReadOnlyObjectWrapper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.scene.web.WebEngine$LoadWorker.updateState(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.scene.web.WebEngine$LoadWorker.dispatchLoadEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/javafx.scene.web.WebEngine$PageLoadListener.dispatchLoadEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.WebPage.fireLoadEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.WebPage.fwkFireLoadEvent(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.network.URLLoaderBase.twkDidFinishLoading(Native Method)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.network.HTTP2Loader.notifyDidFinishLoading(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.webkit.network.HTTP2Loader.lambda$callBackIfNotCanceled$10(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
	at org.jabref.merged.module@5.3.307/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.3.307/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

  ```
</details>

### PR DESCRIPTION
**Pull request by evanpickett**
Fixes #7706 
DOI links in the preview pane no longer open in the WebView and no longer cause uncaught exceptions as a consequence. Unfortunately wasn't able to find a clean solution to opening a browser window on all OS's without using java.awt

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
Hi, thanks for your PR. You can use `JabRefDesktop::openBrowser` to open a url instead of directly using awt-methods.
I think openBrowser uses awt in the end too, but it's at least a common method which behaviour can be changed eventually.

**Comment by evanpickett** [comment]
> Hi, thanks for your PR. You can use `JabRefDesktop::openBrowser` to open a url instead of directly using awt-methods. I think openBrowser uses awt in the end too, but it's at least a common method which behaviour can be changed eventually.

Got it, I went ahead and re-added that functionality then

**Comment by calixtus** [comment]
Great, thanks. Looks good to me! Looking forward. To see more from you eventually!

**Review by calixtus — APPROVED** [review]

File: src/main/java/org/jabref/gui/preview/PreviewViewer.java [hunk 1]
```diff
@@ -137,6 +144,26 @@ public PreviewViewer(BibDatabaseContext database, DialogService dialogService, S
...
+                Node node = anchorList.item(i);
+                EventTarget eventTarget = (EventTarget) node;
+                eventTarget.addEventListener("click", evt -> {
+                    EventTarget target = evt.getCurrentTarget();
+                    HTMLAnchorElement anchorElement = (HTMLAnchorElement) target;
+                    String href = anchorElement.getHref();
+                    try {
+                        JabRefDesktop.openBrowser(href);
+                    } catch (MalformedURLException exception) {
+                        exception.printStackTrace();
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/preview/PreviewViewer.java:160]: Please use the logger instead.

**Review by calixtus — CHANGES_REQUESTED** [review]
Just overlooked an issue. Sorry. 😬

**Comment by Siedlerchr** [comment]
@evanpickett  Would be cool if you could fix the little remaining issue, just use LOGGER.error() so we can merge this

**Comment by evanpickett** [comment]
Sorry I didn't respond quicker, let me know if you need any more changes

**Review by Siedlerchr — APPROVED** [review]
Thanks for the fast follow up. Just wanted to get this in! 😉

**Comment by mlep** [comment]
Fixed in JabRef 5.4--2021-11-25--7aaf5e2
Thank you @evanpickett

**Comment by ThiloteE** [comment]
hey, i just checked out the new development version and it seems this fix not only removed the functionality from entry preview, but also from the main table. Was it desired to remove this as well? Prior to this, i could double click on DOI or URL there and it would open my browser. It was quite convenient, but if this leads to uncaught exceptions...

JabRef 5.4--2021-11-25--7ec4c7e
Windows 10 10.0 amd64 
Java 16.0.2 
JavaFX 17.0.1+1

**Pull request merged**